apps/prairielearn/src/api/trpc/routers/course-files/updateQuestionFiles.ts

```diff
@@ -0,0 +1,64 @@
+import { z } from 'zod';
+
+import { IdSchema } from '../../../../lib/db-types.js';
+import { QuestionModifyEditor } from '../../../../lib/editors.js';
+import { selectCourseById } from '../../../../models/course.js';
+import { privateProcedure, selectUsers } from '../../trpc.js';
+
+export const updateQuestionFiles = privateProcedure
+  .input(
+    z.object({
+      // Context.
+      course_id: IdSchema,
+      user_id: IdSchema,
+      authn_user_id: IdSchema,
+      has_course_permission_edit: z.boolean(),
+      qid: IdSchema,
+
+      // Question data.
+      files: z.record(z.string()),
+    }),
+  )
+  .output(
+    z.object({
+      status: z.union([z.literal('success'), z.literal('error')]),
+      job_sequence_id: z.string(),
+    }),
+  )
+  .mutation(async (opts) => {
+    const course = await selectCourseById(opts.input.course_id);
+
+    const { user, authn_user } = await selectUsers({
+      user_id: opts.input.user_id,
+      authn_user_id: opts.input.authn_user_id,
+    });
+
+    const editor = new QuestionModifyEditor({
+      locals: {
+        authz_data: {
+          has_course_permission_edit: opts.input.has_course_permission_edit,
+          authn_user,
+        },
+        course,
+        user,
+        question: { qid: opts.input.qid },
```

Comment:
@jlevine18 this isn't great; `QuestionModifyEditor` expects a full question. You should use `selectQuestionById` to get a full question, just like we do with `selectCourseById` above.

Reply:
ok, was checking what calls the function made and didn't realize the rest of the expectation. done in b1aff1c15e58f17084559b88c694fd933a09cc3f.